infra/modules/services/diagnosticsettings.bicep

```
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// This template is used to setup diagnostic settings.
+targetScope = 'resourceGroup'
+
+// Parameters
+param loganalyticsName string
+param synapseName string
+param synapseSqlPoolName string
+param synapseSparkPoolName string
+
+//Resources
+resource synapseworkspace 'Microsoft.Synapse/workspaces@2021-06-01' existing = {
+  name: synapseName
+}
+
+resource synapsesqlpool 'Microsoft.Synapse/workspaces/sqlPools@2021-06-01' existing = {
+  parent: synapseworkspace
+  name: synapseSqlPoolName
+}
+
+resource synapsebigdatapool 'Microsoft.Synapse/workspaces/bigDataPools@2021-06-01' existing = {
+  parent: synapseworkspace
+  name: synapseSparkPoolName
+}
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2020-08-01' existing = {
+  name: loganalyticsName
+}
+
+resource diagnosticSetting1 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
```

Let's be consistent here. 👍 
```suggestion
resource diagnosticSetting001 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
```